assignments/life_expectancy/cleaning.py

```diff
@@ -1,75 +1,69 @@
+        rename_raw_cols: Dict[str, str],
+        unpivot_col_id_vars: List[str],
+        rename_unpivot_cols: Dict[str, str],
+        types_unpivot_cols: Dict[str, type],
+        representations_of_nan: List[str],
+        location_col: str,
+        location_filter: str
+) -> pd.DataFrame:
     """
-    Main Script to clean the data.
-
-    It extracts data from .csv, transforms and saves to a new .csv
-
-    Currently it only supports filtering by one region at a time.
-
-    :return: None
+    This function cleans and transforms the data
+
+    :param location_col: str
+        name of the col to filter location
+    :param representations_of_nan: list
+        the possible representations od NaN
+    :param types_unpivot_cols: dict
+        dict w/ types per col {col: type}
+    :param rename_unpivot_cols: dict
+        rename cols {old_name: new_name}
+    :param unpivot_col_id_vars: list
+        list w/ names of the columns used as id for unpivot
+    :param rename_raw_cols: dict
+        id columns that we want to rename {old_name: new_name}
+    :param df: pd.Dataframe
+        to clean and transform
+    :param location_filter: str
+        location filter
+    :return: pd.Dataframe
+        transformed
     """
 
-    t_start = time.time()
-
-    # hardcoded because this func cannot support input
-    configs_path = 'life_expectancy/configs/configs.yml'
-
-    # for inputs description, check f_input_configs docstring
-    load_data_path, save_data_path, rename_raw_cols, \
-        unpivot_col_id_vars, rename_unpivot_cols, types_unpivot_cols, \
-        representations_of_nan, location_col = f_input_configs(configs_path)
-
-    log.info("Starting...")
-
-    log.info("~~~~~~~~~ Extract Data ~~~~~~~~~")
-
-    df_input = import_csv_to_pd(
-        load_data_path, sep=',| \t|\t'
-    )
-
-    log.info("~~~~~~~~~ Transform Data ~~~~~~~~~")
-    log.info("Transforming Input Data")
-
-    df_input = df_input.rename(
+    df = df.rename(
         columns=rename_raw_cols
-    ).query(    # filter location to processing only required data
+    ).query(  # filter location to processing only required data
         f'{location_col} == "{location_filter}"'
     )
 
     log.info("Unpivot Data")
 
     df_ = unpivot_for_digit_cols(
-        df_input, unpivot_col_id_vars
+        df, unpivot_col_id_vars
     ).replace(
         representations_of_nan, np.nan
     ).dropna().astype(  # types of variable and value cols
         types_unpivot_cols
     )
     # it should
-    df_["value"] = df_["value"]/10
-
-    df_ = df_.rename(  # rename variable and value columns
-        columns=rename_unpivot_cols
-    )
-    log.info("~~~~~~~~~ Load Data ~~~~~~~~~")
-    save_pd_to_csv(df_, save_data_path)
+    df_["value"] = df_["value"] / 10
 
-    log.info("~~~~~~~~~ Finished ~~~~~~~~~")
-    log.info('Execution time: %s', round(time.time() - t_start, 3))
+    return df_.rename(columns=rename_unpivot_cols)
```

Comment:
In the clean_data function the dataframe object names could be improved. Especially when the unpivoted dataframe (df_) and pivoted dataframe (df) are almost identically named.

Reply:
Yup. I agree with u. I will replace df_ by df_cl (cl=clean). Tks